### ISSUE TITLE
Rescore doesn't support multiple queries

### ISSUE THREAD

**Issue by gmoskovicz** [elastic/elasticsearch-net#2088]
Rescore implementation currently only support a single query. When you use the fluent API you can only add a single rescore query. If you add multiple queries, a single one will be added.

Easy repro:

```
			// NameScoreFunction its just a customer FunctionScoreFunctionBase implementation

			var searchresults =
				client.Search<Session>(q => q
					  .Query(mainQuery)
					  .Rescore(r => r.RescoreQuery(rq => rq
								.Query(qq => qq
										.FunctionScore(fs => fs
											.Functions(new[] { new NameScoreFunction() { QueryName = "fullname", Field = "fullName" } })
								))
								.QueryWeight(0)
								.RescoreQueryWeight(1))
								.WindowSize(200)
							)
							.Rescore(r2 => r2
								.RescoreQuery(rq => rq
									.Query(qq => qq
										.FunctionScore(fs => fs
											.Functions(new[] { new NameScoreFunction() { QueryName = "company name", Field = "companyNames" } 
									})))
									.QueryWeight(.5)
									.RescoreQueryWeight(.5))
								.WindowSize(200)));
```

The result of this is the following: 

```
{
	"rescore": {
		"window_size": 200,
		"query": {
			"rescore_query": {
				"function_score": {
					"functions": [{
						"name_score": {
							"field": "companyNames",
							"query_name": "company name"
						}
					}]
				}
			},
			"query_weight": 0.5,
			"rescore_query_weight": 0.5
		}
	},
	"query": {
		"match": {
			"primary_name": {
				"query": "Joe Shmo"
			}
		}
	}
}
```

Note that the field is `companyNames` which comes from the second rescore query.

**Comment by russcam** [comment]
Thanks for submitting @gmoskovicz :+1: 

This is a bug; you can work around it for the moment using the low level client and specifying the query using an anonymous type

```c#
	var rs = client.LowLevel.Search<Project>("index", "type", new
	{
		query = new
		{
			match = new
			{
				field1 = new
				{
					@operator = "or",
					query = "the quick brown",
					type = "boolean"
				}
			}
		},
        rescore = new object[] {
			new {
				window_size = 100,
				query = new {
					rescore_query = new {
						match = new {
						   field1 = new {
							  query = "the quick brown",
							  type = "phrase",
							  slop = 2
						   }
						}
					},
					query_weight = 0.7,
					 rescore_query_weight = 1.2
				}
			},
			new {
				window_size = 10,
				query = new {
					score_mode = "multiply",
					rescore_query = new {
						function_score = new {
							script_score = new {
								script = "log10(doc['numeric'].value + 2)"
							}
						}
					},
					query_weight = 0.7,
					rescore_query_weight = 1.2
				}
			}
		 }
	});
```

which yields the query DSL

```javascript
POST http://localhost:9200/index/type/_search 
{
  "query": {
    "match": {
      "field1": {
        "operator": "or",
        "query": "the quick brown",
        "type": "boolean"
      }
    }
  },
  "rescore": [
    {
      "window_size": 100,
      "query": {
        "rescore_query": {
          "match": {
            "field1": {
              "query": "the quick brown",
              "type": "phrase",
              "slop": 2
            }
          }
        },
        "query_weight": 0.7,
        "rescore_query_weight": 1.2
      }
    },
    {
      "window_size": 10,
      "query": {
        "score_mode": "multiply",
        "rescore_query": {
          "function_score": {
            "script_score": {
              "script": "log10(doc['numeric'].value + 2)"
            }
          }
        },
        "query_weight": 0.7,
        "rescore_query_weight": 1.2
      }
    }
  ]
}
```

**Comment by russcam** [comment]
Fixed with #2095. Will go into next 2.x release
